lsp/completion.go

```diff
@@ -0,0 +1,117 @@
+package lsp
+
+import (
+	"os"
+	"strings"
+
+	"github.com/tliron/glsp"
+	protocol "github.com/tliron/glsp/protocol_3_16"
+)
+
+var (
+	snippetKind      = protocol.CompletionItemKindSnippet
+	insertTextFormat = protocol.InsertTextFormatSnippet
+
+	verbCompletionItem = protocol.CompletionItem{
+		Label:      "ftl:verb",
+		Kind:       &snippetKind,
+		Detail:     stringPtr("FTL Verb Annotation"),
+		InsertText: stringPtr("type ${1:request} struct {}\n\ntype ${2:response} struct{}\n\n//ftl:verb\nfunc ${3:name}(ctx context.Context, req ${1:request}) (${2:response}, error) {\n\treturn ${2:response}{}, nil\n}"),
+		Documentation: &protocol.MarkupContent{
+			Kind:  protocol.MarkupKindMarkdown,
+			Value: "Snippet for defining a verb function.\n\n```go\nfunc Name(ctx context.Context, req Request) (Response, error) {}\n```",
+		},
+		InsertTextFormat: &insertTextFormat,
+	}
+
+	enumTypeCompletionItem = protocol.CompletionItem{
+		Label:      "ftl:enum (type)",
+		Kind:       &snippetKind,
+		Detail:     stringPtr("FTL Enum Annotation"),
+		InsertText: stringPtr("//ftl:enum\ntype ${1:enum} string\n\nconst (\n\t${2:Value1} ${1:enum} = \"${2:Value1}\"\n\t${3:Value2} ${1:enum} = \"${3:Value2}\"\n)"),
+		Documentation: &protocol.MarkupContent{
+			Kind:  protocol.MarkupKindMarkdown,
+			Value: "Snippet for defining a type enum.\n\n```go\nconst (\n\tValue1 enum = \"Value1\"\n\tValue2 enum = \"Value2\"\n)\n```",
+		},
+		InsertTextFormat: &insertTextFormat,
+	}
+
+	enumValueCompletionItem = protocol.CompletionItem{
+		Label:      "ftl:enum (value)",
```

Comment:
What do we think about the labels being `//ftl:...` rather than `ftl:...`? It would more accurately reflect the outcome, but is it less discoverable? Like if you just type "ftl" will it still show up? If it does I think that would be preferable, WDYT?

Reply:
Weird things seems to happen when I use `//` to start the auto complete. for example if I type `/` and then select the `//ftl:verb` I get `///ftl:verb`.

I'm not 100% sure how to trigger it to remove the first `/`. It's almost like comments are being treated differently here.

Reply:
In my mind, it feels more like I'm asking for FTL completions if I start typing with `f` vs. a comment with `/`. But, I might feel different for annotations that we're comments style like `@Ingress` of something for kotlin. When using this code, it seems like `f(tl)` is more natural, but I'm def open here.

Reply:
Does it work if you type `//f<tab>` and remove the `//` from the inserted text?

Reply:
I can get it to replace with `//f` but the auto complete doesn't show up. We might have to have this setting turned on in VSCode to get it to work with comments, which I believe is `off` by default.
```json5
"editor.quickSuggestions": {
    "other": "on",
    "comments": "on", //this one.
    "strings": "off"
  },
```